openassessment/xblock/submission_mixin.py

```diff
@@ -288,10 +327,46 @@ def create_submission(self, student_item_dict, student_sub_data):
 
         return submission
 
+    def collect_files_for_submission(self, student_sub_dict):
+        """ Collect files from CSM for individual submisisons or SharedFileUpload for team submisisons. """
+
+        if not self.file_upload_type:
+            return
+
+        for field in ('file_keys', 'files_descriptions', 'files_names', 'files_sizes'):
+            student_sub_dict[field] = []
+
+        if self.teams_enabled:
+            for team_file_upload in self.get_team_files():
+                student_sub_dict['file_keys'].append(team_file_upload.file_key)
+                student_sub_dict['files_descriptions'].append(team_file_upload.description)
+                student_sub_dict['files_names'].append(team_file_upload.item_id)
+                student_sub_dict['files_sizes'].append(team_file_upload.size)
+
+        else:
+            for file_upload in self.file_manager.get_uploads():
+                student_sub_dict['file_keys'].append(file_upload.key)
+                student_sub_dict['files_descriptions'].append(file_upload.description)
+                student_sub_dict['files_names'].append(file_upload.name)
+                student_sub_dict['files_sizes'].append(file_upload.size)
```

Comment:
This could be shortened as:
```python
if self.teams_enabled:
    uploads = self.get_team_files()
else:
    uploads = self.get_uploads()
for upload in uploads:
    student_sub_dict['file_keys'].append(upload.file_key)
    student_sub_dict['files_descriptions'].append(upload.description)
    student_sub_dict['files_names'].append(upload.item_id)
    student_sub_dict['files_sizes'].append(upload.size)
```

Reply:
There's slight key differences between the two upload types (e.g. `file_upload.name` correlates to `team_file_upload.item_id`) which requires the difference

Reply:
ah, right, that's unfortunate.  I've actually got some code in my open PR that will make this easier, so let's just leave it for now, we can do a quick follow-on PR later.

Reply:
if you're tolerant to a healthy dose of _clever_, you could avoid altering the model objects and use the builtin [`getattr`](https://docs.python.org/3/library/functions.html#getattr) to make compound statements when populating the `student_sub_dict`.